### PR TITLE
Add histograms for age of synced checkpoint summaries and contents

### PR DESCRIPTION
**Pull request by mystenmark**

### CODE REVIEW

File: crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs [hunk 1]
```diff
@@ -149,7 +154,14 @@ impl CheckpointExecutor {
...
                 }
                 // Check for newly synced checkpoints from StateSync.
-                _ = self.mailbox.recv() => (),
+                Ok(checkpoint) = self.mailbox.recv() => {
+                    SystemTime::now().duration_since(checkpoint.summary.timestamp())
+                        .map(|latency|
+                            self.metrics.checkpoint_contents_age_ms.report(latency.as_millis() as u64)
+                        )
+                        .tap_err(|err| warn!("unable to compute checkpoint age: {}", err))
+                        .ok();
```
Comment (longbowlu) [inline crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs:163]: can we add a log here with checkpoint height

File: crates/sui-network/src/state_sync/mod.rs [hunk 2]
```diff
@@ -869,6 +869,12 @@ where
             VerifiedCheckpoint::new(checkpoint, &committee).map_err(|(_, e)| e)?
         };
 
+        SystemTime::now()
+            .duration_since(checkpoint.summary.timestamp())
+            .map(|latency| metrics.report_checkpoint_summary_age(latency))
```
Comment (longbowlu) [inline crates/sui-network/src/state_sync/mod.rs:874]: ditto for log

File: crates/sui-network/src/state_sync/metrics.rs [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub(super) struct Metrics(Option<Arc<Inner>>);
 
+impl std::fmt::Debug for Metrics {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        fmt.debug_struct("Metrics").finish()
+    }
+}
+
```
Comment (longbowlu) [inline crates/sui-network/src/state_sync/metrics.rs:19]: why is this needed?

**Review by longbowlu — APPROVED** [review]

File: crates/sui-network/src/state_sync/metrics.rs (hunk 3 above)
Comment (mystenmark) [reply crates/sui-network/src/state_sync/metrics.rs:19]: Histogram doesn't have a Debug impl, so I had to remove it from Inner.

File: crates/sui-network/src/state_sync/mod.rs (hunk 2 above)
Comment (mystenmark) [reply crates/sui-network/src/state_sync/mod.rs:875]: done

File: crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs (hunk 1 above)
Comment (mystenmark) [reply crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs:167]: done

**Review by williampsmith — APPROVED** [review]
